pkg/operator/controller/garden/components.go

```diff
@@ -75,6 +81,50 @@ func (r *Reconciler) newGardenerResourceManager(garden *operatorv1alpha1.Garden,
 	)
 }
 
+func (r *Reconciler) newVirtualGardenGardenerResourceManager(garden *operatorv1alpha1.Garden, secretsManager secretsmanager.Interface) (resourcemanager.Interface, error) {
+	image, err := r.ImageVector.FindImage(images.ImageNameGardenerResourceManager)
+	if err != nil {
+		return nil, err
+	}
+
+	repository, tag := image.String(), version.Get().GitVersion
+	if image.Tag != nil {
+		repository, tag = image.Repository, *image.Tag
+	}
+	image = &imagevector.Image{Repository: repository, Tag: &tag}
+
+	cfg := resourcemanager.Values{
+		AlwaysUpdate:                         pointer.Bool(true),
+		HealthSyncPeriod:                     &metav1.Duration{Duration: time.Minute},
+		Image:                                image.String(),
+		LogLevel:                             r.Config.LogLevel,
+		LogFormat:                            r.Config.LogFormat,
+		MaxConcurrentTokenInvalidatorWorkers: pointer.Int(5),
+		MaxConcurrentTokenRequestorWorkers:   pointer.Int(5),
+		MaxConcurrentCSRApproverWorkers:      pointer.Int(5),
+		NamePrefix:                           namePrefix,
+		PriorityClassName:                    v1beta1constants.PriorityClassNameGardenSystem500,
+		SecretNameServerCA:                   operatorv1alpha1.SecretNameCARuntime,
+		SyncPeriod:                           &metav1.Duration{Duration: time.Minute},
+		TargetDiffersFromSourceCluster:       true,
+		TargetDisableCache:                   pointer.Bool(true),
+		KubernetesVersion:                    r.RuntimeVersion,
+		VPA: &resourcemanager.VPAConfig{
+			MinAllowed: corev1.ResourceList{
+				corev1.ResourceMemory: resource.MustParse("30Mi"),
+			},
+		},
+		WatchedNamespace: pointer.String(r.GardenNamespace),
+	}
+
+	return resourcemanager.New(
+		r.RuntimeClientSet.Client(),
+		r.GardenNamespace,
+		secretsManager,
+		cfg,
+	), nil
```

Comment:
This function is still quite similar to the `botanist.DefaultResourceManager`. Perhaps we can reuse the same parts better?

Reply:
done 😄